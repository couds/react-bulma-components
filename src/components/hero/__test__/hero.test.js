import React from 'react';
import renderer from 'react-test-renderer';
import Hero from '..';
import CONSTANTS from '../../../constants';

describe('Hero component', () => {
  it('Should exist', () => {
    expect(Hero).toMatchSnapshot();
  });
  it('Should expose Hero head, body and footer', () => {
    expect(Hero.Head).toMatchSnapshot();
    expect(Hero.Body).toMatchSnapshot();
    expect(Hero.Footer).toMatchSnapshot();
  });
  it('Should have hero classname', () => {
    const component = renderer.create(
      <Hero>
        Test <a>Give me</a>
      </Hero>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Hero className="other-class this-is-a-test">
        <p>Default</p>
      </Hero>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Hero style={{ height: 250 }}>
        <p>Default</p>
      </Hero>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use gradient style', () => {
    const component = renderer.create(
      <Hero color="primary" gradient>
        <p>Default</p>
      </Hero>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render Hero with hero head, body and footer', () => {
    const component = renderer.create(
      <Hero size="fullheight" color="primary">
        <Hero.Head renderAs="header">
          <div className="bd-notification is-info">Header</div>
        </Hero.Head>
        <Hero.Body>Body</Hero.Body>
        <Hero.Footer renderAs="footer">
          <div className="bd-notification is-danger">Footer</div>
        </Hero.Footer>
      </Hero>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  [null]
    .concat(Object.keys(CONSTANTS.COLORS).map(key => CONSTANTS.COLORS[key]))
    .map(color =>
      it(`Should use use color ${color}`, () => {
        const component = renderer.create(
          <Hero color={color}>
            <p>Default</p>
          </Hero>,
        );
        expect(component.toJSON()).toMatchSnapshot();
      }),
    );
});
