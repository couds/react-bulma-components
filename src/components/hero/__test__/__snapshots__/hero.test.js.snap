// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component Should concat classname in props with Bulma classname 1`] = `
<section
  className="hero other-class this-is-a-test"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should exist 1`] = `[Function]`;

exports[`Hero component Should expose Hero head, body and footer 1`] = `[Function]`;

exports[`Hero component Should expose Hero head, body and footer 2`] = `[Function]`;

exports[`Hero component Should expose Hero head, body and footer 3`] = `[Function]`;

exports[`Hero component Should have hero classname 1`] = `
<section
  className="hero"
>
  Test 
  <a>
    Give me
  </a>
</section>
`;

exports[`Hero component Should render Hero with hero head, body and footer 1`] = `
<section
  className="hero is-primary is-fullheight"
>
  <header
    className="hero-head"
  >
    <div
      className="bd-notification is-info"
    >
      Header
    </div>
  </header>
  <div
    className="hero-body"
  >
    Body
  </div>
  <footer
    className="hero-foot"
  >
    <div
      className="bd-notification is-danger"
    >
      Footer
    </div>
  </footer>
</section>
`;

exports[`Hero component Should use gradient style 1`] = `
<section
  className="hero is-primary is-bold"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use inline styles 1`] = `
<section
  className="hero"
  style={
    Object {
      "height": 250,
    }
  }
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color black 1`] = `
<section
  className="hero is-black"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color danger 1`] = `
<section
  className="hero is-danger"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color dark 1`] = `
<section
  className="hero is-dark"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color info 1`] = `
<section
  className="hero is-info"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color light 1`] = `
<section
  className="hero is-light"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color link 1`] = `
<section
  className="hero is-link"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color null 1`] = `
<section
  className="hero"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color primary 1`] = `
<section
  className="hero is-primary"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color success 1`] = `
<section
  className="hero is-success"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color warning 1`] = `
<section
  className="hero is-warning"
>
  <p>
    Default
  </p>
</section>
`;

exports[`Hero component Should use use color white 1`] = `
<section
  className="hero is-white"
>
  <p>
    Default
  </p>
</section>
`;
